src/main/java/org/jfrog/hudson/pipeline/common/executors/MavenGradleEnvExtractor.java

```diff
@@ -15,8 +15,9 @@
  */
 public class MavenGradleEnvExtractor extends EnvExtractor {
 
-    public MavenGradleEnvExtractor(Run build, BuildInfo buildInfo, Deployer publisher, Resolver resolver, TaskListener buildListener, Launcher launcher, FilePath tempDir, EnvVars env) {
+    public MavenGradleEnvExtractor(Run build, BuildInfo buildInfo, Deployer publisher, Resolver resolver, TaskListener buildListener, Launcher launcher, FilePath tempDir, EnvVars env, boolean skipEncryption) {
         super(build, buildInfo, publisher, resolver, buildListener, launcher, tempDir, env);
+        this.skipEncryption = skipEncryption;
```

Comment:
This is Go-style construction.

Let's create another constructor in EnvExtractor that gets the "boolean skipEncryption" parameter.